Makefile

```diff
@@ -1,12 +1,22 @@
 all: test
 
-uninstall:
-	pip freeze | grep -v "^-e" | xargs pip uninstall -y
-
-install:
+install.tools:
 	pip install --upgrade pip twine wheel
+
+## Install openfisca-core for deployment and publishing.
+install.build: setup.py clean install.tools
+	@## This allows us to be sure tests are run against the packaged version
+	@## of openfisca-core, the same we put in the hands of users and reusers.
+	pip install --upgrade pip wheel
+	python $< bdist_wheel
+	find dist -name "*.whl" -exec pip install --force-reinstall {}[dev] \;
```

Comment:
Why do we reinstall with `dev` dependencies if this is supposed to be the “packaged” version? 🤔

Reply:
Wouldn't the `find dist -name "*.whl"` step more appropriately expressed as `find dist -name "*.whl" -exec rm` in the `clean` target, and `pip install` run after the `clean` when we want to publish?

Reply:
Because we're doing it wrong at the outset: we're using `pip install .[x]` to install dependencies, which is actually to install `openfisca core` **and** the dependencies. Those need to be separate steps.

Reply:
Fixed it, replaced with a more proper -yet not perfect- way of doing it, in the absence of requirement files:

```
python setup.py egg_info  # Create a requirements file.
pip install `grep -v "^\[" *.egg-info/requires.txt`  # Install openfisca's dependencies.
```

Note that this is going to install `openfisca-core` via `country-template` and `extension-template`, so we're forced to reinstall `openfisca-core`.

But, we will reinstall **only** `openfisca-core`:

```
python setup.py bdist_wheel
find dist -name "*.whl" -exec pip install --force-reinstall --no-dependencies {} \;
```